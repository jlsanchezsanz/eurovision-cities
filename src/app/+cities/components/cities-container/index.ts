export * from './cities-container.component';
