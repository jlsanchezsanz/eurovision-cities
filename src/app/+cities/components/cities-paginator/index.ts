export * from './cities-paginator.component';
