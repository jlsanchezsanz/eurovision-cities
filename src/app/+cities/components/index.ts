export * from './cities-container';
export * from './cities-table';
