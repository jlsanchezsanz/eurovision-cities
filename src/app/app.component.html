<header>
  <app-header></app-header>
</header>

<main class="main">
  <section>
    <router-outlet></router-outlet>
  </section>
</main>
