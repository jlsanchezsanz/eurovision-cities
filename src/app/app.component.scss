:host {
  display: flex;
  flex-direction: column;
  padding: 20px;
}

.main {
  margin-top: 80px;
}
