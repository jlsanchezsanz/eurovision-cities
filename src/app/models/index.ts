export * from './cities-response.model';
export * from './city.model';
