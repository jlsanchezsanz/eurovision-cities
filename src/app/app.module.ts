import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { reducers, metaReducers } from './state/reducers';
import { environment } from '../environments/environment';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    AppRoutingModule,
    EffectsModule.forRoot(),
    HttpClientModule,
    StoreModule.forRoot(reducers, {
      metaReducers,
      runtimeChecks: {
        strictStateImmutability: true,
        strictActionImmutability: true,
      },
    }),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    BrowserAnimationsModule,
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
