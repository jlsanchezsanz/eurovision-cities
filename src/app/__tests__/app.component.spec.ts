import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { MockComponent } from 'ng-mocks';

import { AppComponent } from '../app.component';
import { HeaderComponent } from '../components';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent, MockComponent(HeaderComponent)],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.debugElement.componentInstance;
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
