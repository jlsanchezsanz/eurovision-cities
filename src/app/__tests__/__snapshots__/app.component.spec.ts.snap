// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<app-root>
  <header>
    <app-header />
  </header><main
    class="main"
  >
    <section>
      <router-outlet />
    </section>
  </main>
</app-root>
`;
